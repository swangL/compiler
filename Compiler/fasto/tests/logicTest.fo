fun int main() =
  // literals
  let w = write(true) in
  let w = write(false) in
  
  // not
  let w = write(not true) in
  let w = write(not false) in
  let w = write(not not true) in
  let w = write(not not false) in
  
  // and
  let w = write(true && true) in 
  let w = write(true && false) in
  let w = write(false && true) in
  let w = write(false && false) in
  
  // or
  let w = write(true || true) in
  let w = write(true || false) in
  let w = write(false || true) in
  let w = write(false || false) in

  // precedence
  // here the result would be true if '==' had binded weaker than the operators
  // but the result should be false
  let w = write(true == false && not false && false == true) in
  0
  