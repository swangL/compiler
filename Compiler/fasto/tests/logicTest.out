true
false
false
true
true
false
true
false
false
false
true
true
true
false
false